### PR TITLE
TM-4819-FE-Org-Stat

### PR DESCRIPTION
**Pull request by cagbakwu**
[Ticket](https://metaphase.atlassian.net/browse/TM-4918)
[Mock](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/432)
[BE](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/927)

Linked API Endpoints to FE

Image from DEV1
Screenshot 2024-02-15 at 2 49 00 PM

### CODE REVIEW

File: src/Components/AdministratorPage/OrgStats/OrgStats.jsx [hunk 1]
```diff
@@ -65,12 +64,14 @@ const OrgStats = () => {
...
       selectedCycles,
     ];
+
     if (filters.flat().length === 0) {
       setClearFilters(false);
     } else {
       setClearFilters(true);
     }
     dispatch(orgStatsFetchData(getQuery()));
+    dispatch(orgStatsFiltersFetchData());
```
Comment (M00staff) [inline src/Components/AdministratorPage/OrgStats/OrgStats.jsx:74]: `fetchAndSet` gets called every time a filter is selected - should it also be making the call to get the filters?

prolly just need to call `dispatch(orgStatsFiltersFetchData());`, once, on page load

File: src/Components/AdministratorPage/OrgStats/OrgStats.jsx [hunk 2]
```diff
@@ -1,27 +1,23 @@
...
+import { orgStatsFetchData, orgStatsFiltersFetchData, saveOrgStatsSelections } from '../../../actions/orgStats';
 
 const OrgStats = () => {
   const dispatch = useDispatch();
 
   // ================= DATA RETRIEVAL =================
 
-  const genericFilters = useSelector(state => state.filters);
-  const genericFilters$ = get(genericFilters, 'filters') || [];
   const genericFiltersIsLoading = useSelector(state => state.filtersIsLoading);
```
Comment (M00staff) [inline src/Components/AdministratorPage/OrgStats/OrgStats.jsx:21]: why is this still here - are you using the genericFilters for something?

File: src/Components/AdministratorPage/OrgStats/OrgStats.jsx [hunk 3]
```diff
@@ -32,6 +28,16 @@ const OrgStats = () => {
...
+  const filtersIsLoading = useSelector(state => state.orgStatsFiltersIsLoading);
+  const orgStatsfilters = useSelector(state => state.orgStatsFilters);
+
+  const bureaus = orgStatsfilters?.bureauFilters;
+  const orgs = orgStatsfilters?.orgFilters;
+  const cycles = orgStatsfilters?.cycleFilters;
+  const bureauOptions = uniqBy(sortBy(bureaus, [(f) => f.description]), 'description');
+  const orgOptions = uniqBy(sortBy(orgs, [(f) => f.description]), 'code');
+  const cycleOptions = uniqBy(sortBy(cycles, [(f) => f.code]), 'code');
   // ================= FILTER/PAGINATION =================
```
Comment (M00staff) [inline src/Components/AdministratorPage/OrgStats/OrgStats.jsx:41]: will there even be pagination on this page?

File: src/Components/AdministratorPage/OrgStats/OrgStats.jsx (hunk 2 above)
Comment (M00staff) [reply src/Components/AdministratorPage/OrgStats/OrgStats.jsx:21]: ah - yea, i think you wanna swap this for the other filtersIsLoading call

image

**Comment by M00staff** [comment]
Yea - so a lot of the boilerplate code seems fine, I don't think the filtering is wired up correctly, though.

I'm not sure what else this page does, outside of a get request & the ability to filter that response - so definitely want to figure that out and get the filtering working. 

I'd say (even though we said not to) try to deploy your branches on dev and see if any of this actually works, as well. 

This will probably take a few rounds to fully look over.

File: src/Components/AdministratorPage/OrgStats/OrgStats.jsx (hunk 3 above)
Comment (cagbakwu) [reply src/Components/AdministratorPage/OrgStats/OrgStats.jsx:41]: Yea...about the pagination, I didn't see anything in the current code or tasks that ask about pagination.

**Comment by M00staff** [comment]
curious - the cycle is highlighted & looks clickable, but doesn't do anything when clicked... was that just how the design looked, or will it eventually be clickable or something?

if its meant to just LOOK clickable - maybe tweak the CSS so it doesn't change the mouse to a pointer 🤔

image

File: src/Components/AdministratorPage/OrgStats/OrgStats.jsx (hunk 3 above)
Comment (M00staff) [reply src/Components/AdministratorPage/OrgStats/OrgStats.jsx:38]: right, so if there is no pagination - why have a comment here that implies "pagination below?"

File: src/Components/AdministratorPage/OrgStats/OrgStats.jsx [hunk 4]
```diff
@@ -1,37 +1,40 @@
...
+  const filtersHasErrored = useSelector(state => state.orgStatsFiltersHasErrored);
+  const filtersIsLoading = useSelector(state => state.orgStatsFiltersIsLoading);
+  const orgStatsfilters = useSelector(state => state.orgStatsFilters);
+
+  const bureaus = orgStatsfilters?.bureauFilters;
+  const orgs = orgStatsfilters?.orgFilters;
+  const cycles = orgStatsfilters?.cycleFilters;
+  const bureauOptions = uniqBy(sortBy(bureaus, [(f) => f.description]), 'description');
+  const orgOptions = uniqBy(sortBy(orgs, [(f) => f.description]), 'code');
+  const cycleOptions = uniqBy(sortBy(cycles, [(f) => f.code]), 'code');
```
Comment (M00staff) [inline src/Components/AdministratorPage/OrgStats/OrgStats.jsx:37]: I THINK there's a neat vanilla JS way to replace Lodash's `uniqBy` by using Map (not to be confused with `.map`)

not asking you to change it - but i think its pretty neat and worth looking into

File: src/Components/AdministratorPage/OrgStats/OrgStats.jsx [hunk 5]
```diff
@@ -40,40 +43,44 @@ const OrgStats = () => {
...
       selectedBureaus,
       selectedOrgs,
       selectedCycles,
     ];
-    if (filters.flat().length === 0) {
-      setClearFilters(false);
-    } else {
-      setClearFilters(true);
+    const a = [];
+    fils.forEach(f => { if (f.length) { a.push(true); } });
```
Comment (M00staff) [inline src/Components/AdministratorPage/OrgStats/OrgStats.jsx:66]: 👍 love the one liner

File: src/Components/AdministratorPage/OrgStats/OrgStats.jsx [hunk 6]
```diff
@@ -165,30 +176,32 @@ const OrgStats = () => {
...
-                'Total POS': bureauSummary.total_pos,
-                'Total Filled': bureauSummary.total_filled,
-                '% Filled': bureauSummary.total_percent,
-                'Overseas POS': bureauSummary.overseas_pos,
-                'Overseas Filled': bureauSummary.overseas_filled,
-                '% Overseas': bureauSummary.overseas_percent,
-                'Domestic POS': bureauSummary.domestic_pos,
-                'Domestic Filled': bureauSummary.domestic_filled,
-                '% Domestic': bureauSummary.domestic_percent,
+                // the alternative zero is to prevent the value from being falsy
```
Comment (M00staff) [inline src/Components/AdministratorPage/OrgStats/OrgStats.jsx:194]: great comment!

File: src/Components/AdministratorPage/OrgStats/OrgStats.jsx [hunk 7]
```diff
@@ -165,30 +176,32 @@ const OrgStats = () => {
...
+                'Bureau: ': bureauSummary?.bureau_short_desc,
```
Comment (M00staff) [inline src/Components/AdministratorPage/OrgStats/OrgStats.jsx:195]: do you need an `|| '0'` here as well?

**Review by M00staff — APPROVED** [review]
just the one thing with the actions i want you to double check and possible update - but otherwise this is good to go... code looks great btw!

gonna go ahead and approve so it doesn't get held up - but please double check the actions before merging

**Pull request merged**